src/languages/swift.js

```diff
@@ -283,6 +288,143 @@ export default function(hljs) {
   };
   TYPE.contains.push(GENERIC_ARGUMENTS);
 
+  // https://docs.swift.org/swift-book/ReferenceManual/Expressions.html#ID552
+  // Prevents element names from being highlighted as keywords.
+  const TUPLE_ELEMENT_NAME = {
+    begin: lookahead(concat(Swift.identifier, /\s*:/)),
+    end: lookahead(/:/),
+    keywords: "_",
+    relevance: 0
+  };
+  // Matches tuples as well as the parameter list of a function type.
+  const TUPLE = {
```

Comment:
This is a lot of parsing, why would not just matching `blah:` work without trying to find `()` pairs?

Reply:
Because there is a lot of `blah:` in Swift.

Tuple elements, function parameters, dictionary key/value pairs, type/supertype relationship, enum/raw type relationship, switch case labels, labeled control flow statements, ...

The current rule is already pretty naive in assuming that every `blah:` between parenthesis is a tuple element name, so I should add more tests to find cases where it breaks down. It should work relatively well, though, as most of the cases I've listed should not appear in parentheses.

Reply:
Please add the test cases you think are needed and then if the tests fails with a simpler `blah:` rule you can throw it back to me to take a closer look at the failing test case.  I'm remain convinced this can be much simplified.

Reply:
I've added a few more tests for `TUPLE`. So far, I didn't detect anything that required changes to the `TUPLE` rule.